compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -384,6 +385,17 @@ impl<'tcx> Inliner<'tcx> {
                 }
 
                 let fn_sig = self.tcx.fn_sig(def_id).instantiate(self.tcx, args);
+
+                // Additionally, check that the body that we're inlining actually agrees
+                // with the ABI of the trait that the item comes from.
+                if let InstanceDef::Item(instance_def_id) = callee.def
+                    && self.tcx.def_kind(instance_def_id) == DefKind::AssocFn
+                    && let instance_fn_sig = self.tcx.fn_sig(instance_def_id).skip_binder()
+                    && instance_fn_sig.abi() != fn_sig.abi()
+                {
+                    return None;
+                }
```

Comment:
Could probably check that errors have been reported, but 🤷

Reply:
I thought we wanted to avoid depending on global state?

Reply:
still figuring out the exact details on that. But we can probably invoke the wfcheck query responsible for this 😆 not really necessary tho.